clang/lib/Driver/Driver.cpp

```diff
@@ -5065,7 +5071,7 @@ class OffloadingActionBuilder final {
 
             ActionList TformInputs{PostLinkAction, TranslateAction};
             auto *ReplaceFilesAction = C.MakeAction<FileTableTformJobAction>(
-                TformInputs, types::TY_Tempfiletable, types::TY_Tempfiletable);
+                TformInputs, types::TY_Filetable, types::TY_Filetable);
```

Comment:
What do you think about only using `TY_Filetable` when we are keeping the device code around, otherwise we use `TY_Tempfiletable`.  This way, cleanup is performed based on type of file only and not dependent on expecting `TY_Filetable` and `TY_Tempfiletable` to be treated exactly the same when not saving the device files.

Reply:
>> What do you think about only using TY_Filetable when we are keeping the device code around, otherwise we use TY_Tempfiletable
Meaning we set FileTableTformJobAction to TY_Filetable when -fsycl-dump-device-code option is passed?

>> This way, cleanup is performed based on `type of file only` and not dependent on expecting TY_Filetable
Can you please clarify what "type of file only" means here?

Reply:
In Compilation.cpp, you are cleaning up based on 2 checks - TY_Filetable and the dump device flag.  If you modify the output type to be TY_Tempfiletable by default, and only use TY_Filetable when dump device is enabled, you can cleanup just based on the TY_Filetable type only.  Just wanted your thoughts on that.

Reply:
Thank you. 
>>This way, cleanup is performed based on type of file only and **not dependent on expecting `TY_Filetable`**

I got confused by this because we still check if the file is of type `TY_Filetable` in `Compilation.cpp`. So wasn't sure what : _**not dependent on expecting `TY_Filetable`**_ meant here.

I'm ok to only use `TY_Filetable` when dump device option is enabled, especially because we don't have to modify existing test.

FYI, after filtering files based on `TY_Filetable` and peeling off all columns except `"Code"` from the table, each of the `.spv` files still need to be saved, so I'm needing the file extension check in `CleanupFile`